packages/website/docs/creating-multi-source-autocompletes.md

```diff
@@ -2,3 +2,375 @@
+
+For now, `plugins` is an empty array, but you'll learn how to create and add plugins for predefined items, recent searches, and Query Suggestions next.
+
+## Adding predefined items
+
+A popular search pattern for autocomplete menus shows predefined search terms as soon as a user clicks on the search bar and before they begin typing anything. This UX provides a guided experience and exposes users to helpful resources or other content you want them to see.
+
+This tutorial describes how to create a plugin to show static, predefined items. In particular, it exposes helpful links the user may want to refer to.
+
+### Creating a predefined items plugin
+
+Begin by creating a `predefinedItemsPlugin.js` file in your `src` directory, with the following code:
+
+```js
+const predefinedItems = [
+  {
+    label: 'Documentation',
+    url: 'https://autocomplete.algolia.com/',
+  },
+  {
+    label: 'GitHub',
+    url: 'https://github.com/algolia/autocomplete.js/tree/next',
+  },
+]
+
+export const predefinedItemsPlugin = {
+  getSources() {
+    return [
+      {
+        getItems({ query }) {
+          return predefinedItems.filter(
+            (item) =>
+              !query || item.label.toLowerCase().includes(query.toLowerCase())
+          );
+        },
+        getItemUrl({ item }) {
+          return item.url;
+        },
+        templates: {
+          // ...
+        },
+        },
+      },
+    ];
+  },
+};
+
+```
+
+Notice that `predefinedItemsPlugin` has a similar signature as any other autocomplete implementation: it uses the [`getSources`](autocomplete-js/#getsources) option to return an array of items to display. Each object in the array defines where to get items using [`getItems`](sources/#getitems).
+
+In this example, [`getItems`](sources/#getitems) returns a filtered array of `predefinedItems`. The code filters the array to return items that match the query, if it exists. If it doesn't, it returns the entire array. **You can return whatever predefined items you like and format them accordingly.** For example, suppose you want to show trending search items instead of helpful links. Then, you can use [`getItems`](sources/#getitems) to retrieve them from another source, [including an asynchronous API](sources#using-dynamic-sources).
+
+The [`getItemUrl`](sources/#getitemurl) function defines how to get the URL of an item. In this case, since it's an attribute on each object in the `predefinedItems` array, you can simply return the attribute. You can use [`getItemUrl`](sources/#getitemurl) to add [keyboard navigation](keyboard-navigation) to the autocomplete menu. Users can scroll through items in the autocomplete menu with the arrow up and down keys. When they hit enter on one of the `predefinedItems` (or any source that includes[`getItemUrl`](sources/#getitemurl)), it opens the URL retrieved from [`getItemUrl`](sources/#getitemurl).
```

Comment:
```suggestion
The [`getItemUrl`](sources/#getitemurl) function defines how to get the URL of an item. In this case, since it's an attribute on each object in the `predefinedItems` array, you can simply return the attribute. You can use [`getItemUrl`](sources/#getitemurl) to add [keyboard navigation](keyboard-navigation) to the autocomplete menu. Users can scroll through items in the autocomplete menu with the arrow up and down keys. When they hit <kbd>Enter</kbd> on one of the `predefinedItems` (or any source that includes[`getItemUrl`](sources/#getitemurl)), it opens the URL retrieved from [`getItemUrl`](sources/#getitemurl).
```